Upstreamed in 3e37c7d3fa122563a9d88168926f61286ef30cd3

ncurses 6.2 - patch 20200523

this patch differs with upstream:
upstream has integrated this fine in ncurses_dll.h.in (instead of creating a new file)
This change is small, so don't care.  

From ffcd8b69aa8346e94aceefdc4593ad74635708f9 Mon Sep 17 00:00:00 2001
From: Anonymous Maarten <anonymous.maarten@gmail.com>
Date: Sat, 14 Mar 2020 12:38:41 +0100
Subject: [PATCH 08/15] Add include file to add export annotation

---
 configure.in                 |  1 +
 include/headers              |  1 +
 include/ncurses_exports.h.in | 47 ++++++++++++++++++++++++++++++++++++
 3 files changed, 49 insertions(+)
 create mode 100644 include/ncurses_exports.h.in

diff --git a/configure.in b/configure.in
index 2183a978..fc00f9dc 100644
--- a/configure.in
+++ b/configure.in
@@ -2395,6 +2395,7 @@ AC_OUTPUT( \
 	include/MKterm.h.awk \
 	include/curses.head:include/curses.h.in \
 	include/ncurses_dll.h \
+	include/ncurses_exports.h \
 	include/termcap.h \
 	include/unctrl.h \
 	$SUB_MAKEFILES \
diff --git a/include/headers b/include/headers
index dbb134d1..e20798b4 100644
--- a/include/headers
+++ b/include/headers
@@ -33,6 +33,7 @@
 curses.h
 unctrl.h
 ncurses_dll.h
+ncurses_exports.h
 
 # Support for termcap (and tic, etc.), which can be a separate library
 @ termlib
diff --git a/include/ncurses_exports.h.in b/include/ncurses_exports.h.in
new file mode 100644
index 00000000..252f5021
--- /dev/null
+++ b/include/ncurses_exports.h.in
@@ -0,0 +1,47 @@
+/****************************************************************************
+ * Copyright 2020 Thomas E. Dickey                                          *
+ *                                                                          *
+ * Permission is hereby granted, free of charge, to any person obtaining a  *
+ * copy of this software and associated documentation files (the            *
+ * "Software"), to deal in the Software without restriction, including      *
+ * without limitation the rights to use, copy, modify, merge, publish,      *
+ * distribute, distribute with modifications, sublicense, and/or sell       *
+ * copies of the Software, and to permit persons to whom the Software is    *
+ * furnished to do so, subject to the following conditions:                 *
+ *                                                                          *
+ * The above copyright notice and this permission notice shall be included  *
+ * in all copies or substantial portions of the Software.                   *
+ *                                                                          *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS  *
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF               *
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.   *
+ * IN NO EVENT SHALL THE ABOVE COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,   *
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR    *
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR    *
+ * THE USE OR OTHER DEALINGS IN THE SOFTWARE.                               *
+ *                                                                          *
+ * Except as contained in this notice, the name(s) of the above copyright   *
+ * holders shall not be used in advertising or otherwise to promote the     *
+ * sale, use or other dealings in this Software without prior written       *
+ * authorization.                                                           *
+ ****************************************************************************/
+/* $Id: ncurses_exports.h.in,v 1.13 2020/02/29 21:53:14 tom Exp $ */
+
+#ifndef NCURSES_EXPORTS_H_incl
+#define NCURSES_EXPORTS_H_incl 1
+
+#if defined(__CYGWIN__) || defined(_WIN32)
+#  if defined(NCURSES_DLL)
+#    define NCURSES_EXPORT __declspec(dllexport)
+#  elif !defined(NCURSES_STATIC)
+#    define NCURSES_EXPORT __declspec(dllimport)
+#  else
+#    define NCURSES_EXPORT
+#  endif
+#endif
+
+#if !defined(NCURSES_EXPORT)
+#  define NCURSES_EXPORT @NCURSES_EXPORT@
+#endif
+
+#endif /* NCURSES_EXPORTS_H_incl */
-- 
2.21.1

